spec/abstract-jobs.html

```diff
@@ -275,33 +275,17 @@ <h1>AddToKeptObjects ( _object_ )</h1>
     </emu-note>
   </emu-clause>
 
-  <emu-clause id="sec-check-for-empty-cells" aoid="CheckForEmptyCells">
-    <h1> CheckForEmptyCells ( _finalizationRegistry_ ) </h1>
-    <p> The following steps are performed: </p>
-    <emu-alg>
-      1. Assert: _finalizationRegistry_ has an [[Cells]] internal slot.
-      1. For each _cell_ in _finalizationRegistry_.[[Cells]], do
-        1. If _cell_.[[WeakRefTarget]] is ~empty~, then
-          1. Return *true*.
-      1. Return *false*.
-    </emu-alg>
-  </emu-clause>
-
   <emu-clause id="sec-cleanup-finalization-registry" aoid="CleanupFinalizationRegistry">
     <h1> CleanupFinalizationRegistry ( _finalizationRegistry_ [ , _callback_ ] ) </h1>
     <p> The following steps are performed: </p>
     <emu-alg>
-      1. Assert: _finalizationRegistry_ has [[Cells]],
-      [[CleanupCallback]], and [[IsFinalizationRegistryCleanupJobActive]] internal slots.
```

Comment:
I think you still need `[[IsFinalizationRegistryCleanupJobActive]]` to prevent re-entrancy through `cleanupSome`.

Reply:
@mhofman I see how this is an observable difference and we could keep that, but what would be the problem if cleanupSome is called in a reentrant way?

Reply:
It depends on what the original concerns with re-entrancy were.

If it was because the program could end up with 2 active iterators, then yes we don't need this check anymore. But in that case we could have allowed re-entrancy and just correctly tracked the currently "active" iterator instead of throwing on `cleanupSome`.

If it was to avoid having 2 internal iterations over the empty cells, then the re-entrancy check is still needed. It feels like it could be complicated for implementers to support right, I'll let them voice their opinion on this.

Reply:
I don't think the latter was a concern, but it'd be good if implementers told me whether this assumption is valid. I think the concern was really just at the level of, how do the iterators even work if leaked--that you should only be able to iterate on them during the cleanup task, and not before or after. In this sense, [[IsFinalizationRegistryCleanupJobActive]] may have *already* been somewhat overkill.

Reply:
I don't see an issue with recursively calling cleanupSome without iterators to worry about. It's an odd way to iterate through holdings but I don't see any problems really.